### PR TITLE
`models`: Refactors error handling to be "module" specific

### LINKED ISSUE #121: 0.2.0: Do away with crate-wide error modules
Error submodules are preferred for error types, and it is preferable to create multiple error classifications rather than bubbling all errors up to a single error type.

### PR DESCRIPTION
**Pull request by jpmcb**
```
Modifies consumed error handling from API server and controller modules

Signed-off-by: John McBride <jpmmcb@amazon.com>
```

**Issue number:**

Closes: #121

**Description of changes:**

Refactors errors in `models/`to be structured around each module. This also removed a few structured errors that I didn't seem to be used in the codebase anymore.

**Testing done:**

Built image and loaded it into a kind cluster. Looks 👍🏼 

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by jpmcb** [comment]
Force pushed to fixup commit message and remove a WIP commented out line

**Review by gthao313 — APPROVED** [review]
Some conflicts have to be fixed, otherwise LGTM!  :rocket:

File: apiserver/src/api/error.rs [hunk 1]
```diff
@@ -13,10 +13,14 @@ pub enum Error {
     MissingClusterIPFamiliy { source: std::env::VarError },
```
Comment (gthao313) [inline apiserver/src/api/error.rs:13]: It's not relevant to your code change, but can you fix this typo? thanks! 
```MissingClusterIPFamily```

**Comment by jpmcb** [comment]
Force pushed to address merge conflicts

**Review by gthao313 — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Pull request merged**